lib/core/asynctree.js

```diff
@@ -189,7 +189,7 @@ class AsyncTree extends EventEmitter{
       }
 
       const isAssertion = node.parent && (node.parent.namespace === 'assert' || node.parent.namespace === 'verify');
-      if (node.parent && !node.parent.isRootNode && !isAssertion && !stillInProgress) {
+      if (node.parent && !node.parent.isRootNode && !isAssertion && !stillInProgress && node.parent.name !== 'waitUntil') {
```

Comment:
We shouldn't ideally hardcode a command's name over here, but instead have some logic inside `this.otherChildNodesInProgress()` method to decide whether the command has finished completely or there are still some child nodes that can potentially be added to the command, again without hardcoding the command name.

Reply:
Also, if the `waitUntil` command fails, we should ideally abort the test execution and display the error instead of continuing the test execution.

`await browser.waitUntil().pause()` would do just that (if the `waitUntil` command fails, it won't execute the `pause()` command.

But if we write it as:
```js
await browser.waitUntil();
browser.pause();
```
it would execute the `pause()` command even after the `waitUntil` command fails.

Reply:
What we can do here is (for the first comment) declare a flag in the command file itself (as you suggested earlier) using a get method and then use that flag to decide whether to resolve the parent node as well with the same result or not.

This can be done by returning the flag value from `getCommandOptions()` method in `lib/api/_loaders/_command-loader.js` file, which will then be available in `node.options` in the `asynctree.js` file and can be used there.

And instead of using `this.otherChildNodesInProgress()` for this purpose, we can create a new `this.resolveParentNodeWithResult()` method and put the logic for deciding this in there.

Reply:
And then maybe we can iteratively do this for all the commands to deprecate this logic altogether (as this logic I think is only relevant to the client-commands which then call some protocol command to do the work, as we're slowly deprecating such commands).

I think a good way to do this would be to bypass this logic for a few commands in every release to ensure that we don't end up breaking anything (and if we get complain for some commands, we can just revert the commit from that release). And once we've bypassed this logic for all the commands, we can safely remove this.

For the note, according to my understanding, this logic was used for commands that call other commands to do the actual work, and when those "child-commands" returned a result, the parent command (which is the original command that the user used) should also be resolved with the same result, so we resolve the parent node as well with the same result as the child node using this logic.